### PR TITLE
Add migration test for comparing serializations

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: migration/src/test/java/no/unit/nva/datamodel/FunctionalTests.java [hunk 1]
```diff
@@ -93,26 +101,49 @@ public void gradleRunnerRunsSampleProjectOnPreviousVersion() throws FileNotFound
...
     @Test
-    public void previousVersionSerializationsShouldBeDeserializedByCurrentVersion() throws FileNotFoundException {
-        List<String> jsons = listSerializedPublications(sampleProjectPreviousVersion);
-        for (String json : jsons) {
-            assertDoesNotThrow(() -> objectMapper.readValue(json, Publication.class));
+    public void currentVersionShouldProduceSameDeserializationWhenDeserializingOldAndNewSerializedVersions()
+        throws FileNotFoundException {
+        List<String> jsons = listSerializedPublications(samplesProjectPreviousVersion);
+
+        var deserializedFromOldVersionSerialization = jsons.stream()
```
Comment (janbj) [inline migration/src/test/java/no/unit/nva/datamodel/FunctionalTests.java:116]: extract method

File: migration/src/test/java/no/unit/nva/datamodel/FunctionalTests.java [hunk 2]
```diff
@@ -93,26 +101,49 @@ public void gradleRunnerRunsSampleProjectOnPreviousVersion() throws FileNotFound
...
+            .collect(Collectors.toMap(Publication::getIdentifier, publication -> publication));
+
+        var deserializedFromNewVersionSerialization = deserializedFromOldVersionSerialization.values().stream()
+            .map(this::serializeWithCurrentVersion)
+            .map(this::deserializeWithCurrentVersion)
+            .collect(Collectors.toMap(Publication::getIdentifier, publication -> publication));
+
+        var allIdentifiers = deserializedFromOldVersionSerialization.keySet();
+
+        for (SortableIdentifier identifier : allIdentifiers) {
```
Comment (janbj) [inline migration/src/test/java/no/unit/nva/datamodel/FunctionalTests.java:127]: extract method

**Review by brinxmat — APPROVED** [review]

File: migration/src/test/java/no/unit/nva/datamodel/FunctionalTests.java [hunk 3]
```diff
@@ -170,16 +202,56 @@ private static void deleteExistingSerializations(File sampleProjectFolder) throw
...
+            MATCH_DATAMODEL_VERSION_IN_DEPENDENCY_FILE_FOR_REPLACING_IT, datamodelDependency(version));
+        assertThat(dependenciesFileContents,containsString(version));
+        return dependenciesFileContents;
+    }
+
+    private static void writeBackDependenciesFileWithUpdatedDatamodelDependency(File dependenciesFile,
+                                                                                String dependenciesFileContents)
+        throws IOException {
+        Files.delete(dependenciesFile.toPath());
+        BufferedWriter writer = new BufferedWriter(new FileWriter(dependenciesFile));
```
Comment (janbj) [inline migration/src/test/java/no/unit/nva/datamodel/FunctionalTests.java:223]: use try with resource?

File: migration/src/test/java/no/unit/nva/datamodel/FunctionalTests.java [hunk 4]
```diff
@@ -170,16 +202,56 @@ private static void deleteExistingSerializations(File sampleProjectFolder) throw
...
+
+    private static void verifyThatDependenyFileContainsDatamodelDependency(String dependenciesFileContents) {
+        assertThat(DATAMODEL_DEPENDENCY_NOT_FOUND_ERROR,
+                   PATTERN_FOR_FINDING_DATAMODEL_VERSION_IN_DEPENDENCIES_FILE
+                       .matcher(dependenciesFileContents).matches(), is(true)
+        );
+    }
+
+    private static String datamodelDependency(String version) {
+        return String.format("%sdatamodel = {strictly = '%s'}%s",
```
Comment (janbj) [inline migration/src/test/java/no/unit/nva/datamodel/FunctionalTests.java:248]: use template constant?

**Review by janbj — APPROVED** [review]

**Pull request merged**